integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/freeipa/FreeIpaUpgradeTests.java

```diff
@@ -301,10 +306,37 @@ private void kinit(SdxTestDto sdxTestDto, SdxClient sdxClient) {
         }
     }
 
+    private void dnsLookups(SdxTestDto sdxTestDto, SdxClient sdxClient) {
+        String fqdn = null;
+        String privateIP = "";
+        List<InstanceGroupV4Response> instanceGroups = getInstanceGroups(sdxTestDto, sdxClient);
+        for (InstanceGroupV4Response instanceGroup : instanceGroups) {
+            fqdn = instanceGroup.getMetadata().stream().filter(item -> item.getInstanceGroup().equals("idbroker"))
+                    .findFirst().map(item -> item.getDiscoveryFQDN()).orElse(null);
+            privateIP = instanceGroup.getMetadata().stream().filter(item -> item.getInstanceGroup().equals("idbroker"))
+                    .findFirst().map(item -> item.getPrivateIp()).orElse(null);
+            if (fqdn != null) {
+                break;
+            }
+        }
```

Comment:
I understand this a test but could you refactor this? you should be able to use a single stream to find the idbroker instancemetadata, and throw a `TestFailException` if no suitable found

Reply:
Done.